pravega/src/main/scala/akka/stream/alpakka/pravega/impl/PravegaCapabilities.scala

```diff
@@ -0,0 +1,22 @@
+/*
+ * Copyright (C) 2016-2020 Lightbend Inc. <https://www.lightbend.com>
+ */
+
+package akka.stream.alpakka.pravega.impl
+import akka.annotation.InternalApi
+import akka.stream.stage.StageLogging
+import io.pravega.client.ClientConfig
+import io.pravega.client.EventStreamClientFactory
+
+@InternalApi private[pravega] trait PravegaCapabilities {
+
+  this: StageLogging =>
+
+  val scope: String
+  def clientConfig: ClientConfig
```

Comment:
These two fields could be declared as `def t:T` here and later, where they are actually assigned, be overriden as `val t:T`.

For example, I wonder if there's a reason for `clientConfig` to be `def` and for `scope` to be `val.. I checked all the ploaces where `clientConfig` is overriden and I think it's always a `val` so I'm not sure what is the purpose of it being a `def` (probably I'm missing something).